pkg/kv/store.go

```diff
@@ -0,0 +1,97 @@
+package kv
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+)
+
+var (
+	ErrUnknownDriver   = errors.New("unknown driver")
+	ErrOperationFailed = errors.New("operation failed")
+	ErrMissingKey      = errors.New("missing key")
+	ErrMissingValue    = errors.New("missing value")
+	ErrNotFound        = errors.New("not found")
+	ErrClosedEntries   = errors.New("closed entries")
+)
+
+type Entry struct {
+	Key   []byte
+	Value []byte
+}
+
+type Driver interface {
+	// Open access to the database store. migrate kv if needed
+	Open(ctx context.Context, dsn string) (Store, error)
```

Comment:
What is `dsn`?

Reply:
Data Source Name. The short name for connection string.